packages/flutter/lib/src/widgets/tooltip.dart

```diff
@@ -0,0 +1,928 @@
+///
+/// The animation property exposes the underlying tooltip overlay child show and
+/// hide animation. This can be used to drive animations that sync up with the
+/// tooltip overlay child show/hide animation, for example to fade the tooltip
+/// in and out.
+typedef TooltipComponentBuilder =
+    Widget Function(BuildContext context, Animation<double> animation);
+
+/// Signature for computing the position of a tooltip.
+///
+/// The [TooltipPositionContext] contains all the necessary information for
+/// positioning the tooltip, including the target location, sizes, offset, and
+/// positioning preference.
+///
+/// Returns the offset from the top left of the overlay to the top left of the tooltip.
+///
+/// See also:
+///
+///  * [TooltipPositionContext], which contains the positioning parameters.
+typedef TooltipPositionDelegate = Offset Function(TooltipPositionContext context);
+
+/// Contextual information for positioning a tooltip.
+///
+/// This immutable data class contains all the necessary information for computing
+/// the position of a tooltip relative to its target widget.
+///
+/// See also:
+///
+///  * [TooltipPositionDelegate], which uses this context to compute tooltip positions.
+@immutable
+class TooltipPositionContext {
+  /// Creates a tooltip position context.
+  const TooltipPositionContext({
+    required this.target,
+    required this.targetSize,
+    required this.tooltipSize,
+    required this.verticalOffset,
+    required this.preferBelow,
+    this.overlaySize = Size.infinite,
+  });
+
+  /// The center point of the target widget in the global coordinate system.
+  final Offset target;
+
+  /// The size of the target widget that triggers the tooltip.
+  final Size targetSize;
+
+  /// The size of the tooltip itself.
+  final Size tooltipSize;
+
+  /// The size of the overlay within which the tooltip is displayed.
+  final Size overlaySize;
+
+  /// The configured vertical offset.
+  final double verticalOffset;
+
+  /// Whether the tooltip prefers to be positioned below the target.
+  final bool preferBelow;
+
+  @override
+  bool operator ==(Object other) {
+    if (identical(this, other)) {
+      return true;
+    }
+    if (other.runtimeType != runtimeType) {
+      return false;
+    }
+    return other is TooltipPositionContext &&
+        other.target == target &&
+        other.targetSize == targetSize &&
+        other.tooltipSize == tooltipSize &&
+        other.overlaySize == overlaySize &&
+        other.verticalOffset == verticalOffset &&
+        other.preferBelow == preferBelow;
+  }
+
+  @override
+  int get hashCode =>
+      Object.hash(target, targetSize, tooltipSize, overlaySize, verticalOffset, preferBelow);
+}
+
+/// The method of interaction that will trigger a tooltip.
+/// Used in [RawTooltip.triggerMode].
+///
+/// On desktop, a tooltip will be shown as soon as a pointer hovers over
+/// the widget, regardless of the value of [RawTooltip.triggerMode].
```

Comment:
What about on a mobile/touch device with a mouse connected? I'm assuming it behaves the same as desktop. If so maybe reword this "When using a pointer device like a mouse, a tooltip will be..."